src/Servers/Kestrel/test/InMemory.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -323,6 +323,190 @@ await using (var server = new TestServer(context => Task.CompletedTask, new Test
+
+                    connection.ShutdownSend();
+
+                    await connection.ReceiveEnd();
+                }
+            }
+
+            Assert.Null(loggedException);
+            // Verify DiagnosticSource event for bad request
+            Assert.False(badRequestEventListener.EventFired);
+        }
+
+        [Fact]
+        public async Task ExtraLinesIgnoredBetweenAdjacentRequests()
+        {
+            BadHttpRequestException loggedException = null;
+
+            TestSink.MessageLogged += context =>
+            {
+                if (context.EventId.Name == "ConnectionBadRequest" && context.Exception is BadHttpRequestException ex)
+                {
+                    loggedException = ex;
+                }
+            };
+
+            // Set up a listener to catch the BadRequest event
+            var diagListener = new DiagnosticListener("NotBadRequestTestsDiagListener");
+            var badRequestEventListener = new BadRequestEventListener(diagListener, (pair) => { });
+
+            await using (var server = new TestServer(context => context.Request.Body.DrainAsync(default), new TestServiceContext(LoggerFactory) { DiagnosticSource = diagListener }))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.SendAll(
+                        "POST / HTTP/1.1",
+                        "Host:",
+                        "Content-Length: 5",
+                        "",
+                        "funny",
+                        "",
+                        "",
+                        "",
+                        "POST /"); // Split the request line
+
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        "Content-Length: 0",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "",
+                        "");
+
+                    await connection.SendAll(
+                        " HTTP/1.1",
+                        "Host:",
+                        "Content-Length: 5",
+                        "",
+                        "funny");
+
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        "Content-Length: 0",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "",
+                        "");
+
+                    connection.ShutdownSend();
+
+                    await connection.ReceiveEnd();
+                }
+            }
+
+            Assert.Null(loggedException);
+            // Verify DiagnosticSource event for bad request
+            Assert.False(badRequestEventListener.EventFired);
+        }
+
+        [Theory]
+        [InlineData("\r")]
+        [InlineData("\n")]
+        [InlineData("\r\n")]
+        [InlineData("\n\r")]
+        [InlineData("\r\n\r\n")]
+        [InlineData("\r\r\r\r\r")]
```

Comment:
does any system use `\r` as a newline? they are all `\n` or `\r\n` that I know of. I don't see `\n\n` here.

Reply:
The HTTP spec requires `\r\n`, that's not OS specific.

Reply:
It's not so much that any system uses `\r` as a newline, but rather that clients in the wild sometimes have _interesting_ implementations of HTTP. The bare CR was one such example that we encountered.

RFC 7230 encourages some leniency in this area (ignoring empty lines) and this also matches IIS's behavior which makes things smoother for folks migrating over to Kestrel, so we decided this is a safe change.

The `\n\n` case works too, I'll add it.